R/rromeo_functions.R

```diff
@@ -67,18 +67,102 @@ rr_publisher = function(given_id) {
 #' Journal data by ISSN
 #'
 #' @param issn A single journal ISSN
+#'
 #' @export
 #'
 #' @examples
 #'
-#' rr_journal("1947-6264")
-rr_journal = function(issn) {
+#' rr_journal_issn("1947-6264")
+rr_journal_issn = function(issn) {
 
   validate_issn(issn)
 
-  GET(rr_base_api(), query = list(issn = issn))
+  api_answer = GET(rr_base_api(), query = list(issn = issn))
+
+  parse_answer(api_answer, multiple = FALSE)
 }
 
+#' Journal data by title
+#'
+#' @param name A character string, containing the (possibly) partial name of the
+#' journal
+#' @param qtype A character string saying whether you are lookin for `exact`,
+#' `contains` or `starts with` matches
+#' @inheritParams parse_answer
+#'
+#' @export
+#'
+#' @examples
+#' rr_journal_name("Journal of Geology")
+#' rr_journal_name("Biogeography", multiple = FALSE, qtype = "contains")
+#' rr_journal_name("Biogeography", multiple = TRUE, qtype = "contains")
+rr_journal_name = function(name, multiple = FALSE,
+                           qtype = c("exact", "contains", "starts with")) {
+
+  qtype = match.arg(qtype)
+
+  api_answer = GET(rr_base_api(), query = list(jtitle = name, qtype = qtype))
+
+  parse_answer(api_answer, multiple = multiple)
+}
+
+#' Parse API answer.
+#'
+#' @param multiple If multiple results match your query, should the function
+#' recursively fetch data for each of one of them (`multiple = TRUE`) or
+#' return a data.frame containing only titles and ISSN of all matches
+#' (`multiple = FALSE`)
+#'
+#' @keywords internal
+#'
+#' @import xml2
+parse_answer = function(api_answer, multiple = FALSE) {
+
+  if (http_error(api_answer)) {
+    stop("The API endpoint could not be reached. Please try again later.")
+  }
+
+  xml_source = content(api_answer, encoding = "ISO-8859-1")
+
+  hits = xml_text(xml_find_all(xml_source, "//numhits"))
+
+  if (hits == 0) {
+    stop("No journal matches your query terms. Please try another query.")
+  }
+  else if (hits == 1) {
+    romeocolour = xml_text(xml_find_all(xml_source, "//romeocolour"))
```

Comment:
Add title & issn for easier reading of the extracted data
```suggestion
    title = xml_text(xml_find_all(xml_source, "//jtitle"))
    issn = xml_text(xml_find_all(xml_source, "//issn"))
    romeocolour = xml_text(xml_find_all(xml_source, "//romeocolour"))
```